worker/src/main/java/gov/cms/ab2d/worker/processor/PatientClaimsProcessorImpl.java

```diff
@@ -191,7 +192,29 @@ private void handleException(StreamHelper helper, String data, Exception e) thro
                 .filter(Objects::nonNull)
                 // Remove Plan D
                 .filter(resource -> !isPartD(resource))
+                // Make sure the returned patient ID is actually part of the contract
+                .filter(resource -> validPatientInContract(resource, patientsMap))
                 // compile the list
                 .collect(Collectors.toList());
     }
+
+    /**
+     * returns true if the patient is a valid member of a contract, false otherwise. If either value is empty,
+     * it returns false
+     *
+     * @param benefit - The benefit to check
+     * @param patients - the patient map containing the patient id & patient object
+     * @return true if this patient is a member of the correct contract
+     */
+    boolean validPatientInContract(ExplanationOfBenefit benefit, Map<String, ContractBeneficiaries.PatientDTO> patients) {
+        if (benefit == null || patients == null) {
+            return false;
```

Comment:
There should be some sort of logging if there is invalid patients. We would want to know about this and try and notify BFD.

Reply:
Returning false here is most likely a usage issue, not a data issue so just made it a log.debug here, but I added a log.error to the not found one.